tl-services/src/main/java/org/egov/tl/validator/TLValidator.java

```diff
@@ -215,6 +217,62 @@ private void validateInstitution(TradeLicenseRequest request){
     }
 
 
+
+    /**
+     *  Validates the fromDate and toDate of the request
+     * @param request The input TradeLicenseRequest Object
+     */
+    public void validateRenewal(TradeLicenseRequest request){
+        Map<String,String> errorMap = new HashMap<>();
+        
+        TradeLicenseSearchCriteria criteria = new TradeLicenseSearchCriteria();
+        List<String> ids = new LinkedList<>();
+        List<String> licenseNumbers = new LinkedList<>();
+        request.getLicenses().forEach(license -> {
+            ids.add(license.getId());
+            if(license.getLicenseNumber() != null){
+                licenseNumbers.add(license.getLicenseNumber());
+            }            
+        });
+        criteria.setTenantId(request.getLicenses().get(0).getTenantId());
+        criteria.setIds(ids);
+        criteria.setBusinessService(request.getLicenses().get(0).getBusinessService());
+        if(licenseNumbers.size() > 0){
+            criteria.setLicenseNumbers(licenseNumbers);
+        }
+        List<TradeLicense> searchResult = tlRepository.getLicenses(criteria);
+        Map<String , TradeLicense> licenseMap = new HashMap<>();
+        searchResult.forEach(license -> {
+            licenseMap.put(license.getLicenseNumber() , license);
+        });
+        
+        
+        request.getLicenses().forEach(license -> {
+            if(license.getApplicationType() != null && license.getApplicationType().toString().equals(TLConstants.APPLICATION_TYPE_RENEWAL)){
+                if(licenseMap.containsKey(license.getLicenseNumber())){
+                    TradeLicense searchObj = licenseMap.get(license.getLicenseNumber());
+                    Long currentFromDate = license.getValidFrom();
+                    Long currentToDate = license.getValidTo();
+                    Long existingFromDate = searchObj.getValidFrom();
+                    Long existingToDate = searchObj.getValidTo();
+                    if(currentFromDate < existingToDate){
+                        errorMap.put("INVALID FROM DATE","ValidFrom should be greater than the previous applications ValidTo Date");
+                    }
+                    if(currentFromDate  <= existingFromDate){
```

Comment:
we should throw error if currentFromDate < existingToDate   and   currentFromDate > currentToDate